src/hw-monitor.h

```diff
@@ -325,5 +325,14 @@ namespace librealsense
         static std::string get_firmware_version_string(const std::vector<uint8_t>& buff, size_t index, size_t length = 4);
         static std::string get_module_serial_string(const std::vector<uint8_t>& buff, size_t index, size_t length = 6);
         bool is_camera_locked(uint8_t gvd_cmd, uint32_t offset) const;
+
+        template <typename T>
+        T get_gvd_field(const std::vector<uint8_t>& data, size_t index)
+        {
+            T rv = 0;
+            for (int i = 0; i < sizeof(T); i++)
```

Comment:
Pls add compile-time boundary's check

Reply:
compile time check is not possible, run time check added.